app/models/task.rb

```diff
@@ -1,11 +1,16 @@
 class Task < ApplicationRecord
   belongs_to :project
+
   validates :project, presence: true
   validates :title, presence: true
-  validate :deadline_after_creation
+  validate :deadline_later
 
   private
-  def deadline_after_creation
-    errors.add(:deadline_at, "must be after creation time") unless (!created_at.nil? and created_at < deadline_at) or Time.now.to_datetime < deadline_at
+
+  def deadline_later
+
+    unless (!created_at.nil? && created_at < deadline_at) || (created_at.nil? && Time.now.to_datetime < deadline_at)
```

Comment:
Well, when I debugged behaviour of this code segment I came to conclusion that created_at might be nil at this point, is object is not yet saved for the first time

Reply:
Yeah, you are right. Then maybe you should check deadline_at presence too? And it is more preferable to use the reverse method instead of the negative form (`present?` instead of `!...nil?`). And you can replace your created_at presence checks with `(created_at || Time.now)`

Reply:
And can you recode it using "if" to avoid nesting?